tensorly/backend/paddle_backend.py

```diff
@@ -0,0 +1,366 @@
+
+        Example
+        -------
+
+        >>> import tensorly as tl
+        >>> import numpy as np
+        >>> tensor = tl.tensor([[1, 2, 3], [4, 5, 6]])
+        >>> cpy = tensor.copy()
+        >>> tensor[:, 1] = 0
+        >>> tensor
+        array([[1, 0, 3],
+                [4, 0, 6]])
+        >>> tl.index_update(tensor, tl.index[:, 1], 0)
+        array([[1, 0, 3],
+               [4, 0, 6]])
+
+        See also
+        --------
+        index
+        """
+        if paddle.is_tensor(values):
+            if paddle.is_tensor(indices):
+                # indices is bool mask
+                mask_coord = paddle.concat(
+                    paddle.nonzero(indices, as_tuple=True),
+                    axis=1,
+                )
+                t = paddle.scatter_nd_add(
+                    tensor * (~indices).astype(tensor.dtype),
+                    mask_coord,
+                    values,
+                )
+                paddle.assign(t, tensor)  # inplace update
+            else:
+                # NOTE: Assure source Tensor is contiguous
+                tensor[indices] = (
+                    values if values.is_contiguous() else values.contiguous()
+                )
+        else:
+            tensor[indices] = values
+        return tensor
+
+    @staticmethod
+    def to_numpy(tensor: paddle.Tensor):
+        if paddle.is_tensor(tensor):
+            return tensor.numpy()
+        elif isinstance(tensor, np.ndarray):
+            return tensor
+        else:
+            return np.asarray(tensor)
+
+    @staticmethod
+    def shape(tensor: paddle.Tensor):
+        return tuple(tensor.shape)
+
+    @staticmethod
+    def ndim(tensor: paddle.Tensor):
+        return tensor.ndim
+
+    @staticmethod
+    def arange(start, stop=None, step=1.0, *args, **kwargs):
+        if stop is None:
+            return paddle.arange(
+                start=0.0, end=float(start), step=float(step), *args, **kwargs
+            )
+        else:
+            return paddle.arange(
+                float(start), float(stop), float(step), *args, **kwargs
+            )
+
+    @staticmethod
+    def clip(tensor: paddle.Tensor, a_min=None, a_max=None, inplace=False):
+        if inplace:
+            paddle.clip_(tensor, a_min, a_max)
+            return tensor
+        else:
+            return paddle.clip(tensor, a_min, a_max)
+
+    @staticmethod
+    def all(tensor: paddle.Tensor):
+        return paddle.sum(tensor != 0)
+
+    def transpose(self, tensor: paddle.Tensor, axes: int | Sequence[int] | None = None):
+        axes = axes or list(range(self.ndim(tensor)))[::-1]
+        if not isinstance(axes, (tuple, list)):
+            axes = list(axes)
+        return tensor.transpose(axes)
+
+    @staticmethod
+    def copy(tensor: paddle.Tensor):
+        return tensor.clone()
+
+    @staticmethod
+    def norm(tensor: paddle.Tensor, order=None, axis=None):
+        # paddle does not accept `None` for any keyword arguments. additionally,
```

Comment:
This comment is [not consistent with the paddle documentation](https://www.paddlepaddle.org.cn/documentation/docs/en/2.2/api/paddle/norm_en.html#norm). Can you confirm that `norm()` needs to be handled as a special case like this? It looks like you may be able to just import it at the bottom with `eigh()`, etc.

Reply:
You are right, only `order` be changed to float when order is `inf`/`-inf`.